### PR TITLE
Models docs: Improve flavor-specific documentation

### PR DESCRIPTION
**Pull request by dbczumar**
This PR improves the clarity of MLflow's model flavors documentation.

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
 
-The H2O model flavor enables logging and loading H2O models. These models will be saved by using the :py:mod:`mlflow.h2o.save_model`. Using :py:mod:`mlflow.h2o.log_model` will also enable a valid ``Python Function`` flavor.
+The ``h2o`` model flavor enables logging and loading H2O models. 
 
-When loading a H2O model as a PyFunc model, :py:mod:`h2o.init(...)` will be called. Therefore, the right version of h2o(-py) has to be in the environment. The arguments given to :py:mod:`h2o.init(...)` can be customized in ``model.h2o/h2o.yaml`` under the key ``init``. For more information, see :py:mod:`mlflow.h2o`.
+The :py:mod:`mlflow.h2o` module defines :py:func:`save_model() <mlflow.h2o.save_model>` and 
+:py:func:`log_model() <mlflow.h2o.log_model>` methods for saving H2O models in MLflow model format.
+These methods produce MLflow models with the ``python_function`` flavor, allowing them to be loaded 
+as generic Python functions for inference via :py:func:`mlflow.pyfunc.load_pyfunc()`. When MLflow 
+models with the ``h2o`` flavor are loaded via :py:func:`load_pyfunc() <mlflow.pyfunc.load_pyfunc>`,
```
Comment (stbof) [inline docs/source/models.rst]: When MLflow  models with the ``h2o`` flavor are loaded via -> When you load MLflow models with the ``h2o`` flavor are loaded using

File: docs/source/models.rst [hunk 2]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
+
+For more information, see :py:mod:`mlflow.h2o`.
 
 Keras (``keras``)
 ^^^^^^^^^^^^^^^^^
 
-The ``keras`` model flavor enables logging and loading Keras models. This model will be saved in a HDF5 file format, via the model_save functionality provided by Keras. Additionally, model can be loaded back as ``Python Function``. For more information, see :py:mod:`mlflow.keras`.
+The ``keras`` model flavor enables logging and loading Keras models. The :py:mod:`mlflow.keras`
+module defines :py:func:`save_model() <mlflow.keras.save_model>` and 
+:py:func:`log_model() <mlflow.keras.log_model>` functions that can be used to save Keras models
```
Comment (stbof) [inline docs/source/models.rst]: that can be used to save -> that you can use to save

File: docs/source/models.rst [hunk 3]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
+in MLflow model format. These functions serialize Keras models as HDF5 files using the Keras
+library's built-in model persistence functions. MLflow models produced by these functions
+also contain the ``python_function`` flavor, allowing them to be interpreted as generic
+Python functions for inference via :py:func:`mlflow.pyfunc.load_pyfunc()`. Finally, the
+:py:func:`mlflow.keras.load_model` function can be used to load MLflow models with the
```
Comment (stbof) [inline docs/source/models.rst]: Finally, you can use the :py:func:`mlflow.keras.load_model` function ...

File: docs/source/models.rst [hunk 4]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
 
-The ``mleap`` model flavor supports saving models using the MLeap persistence mechanism. A companion module for loading MLflow models with the MLeap flavor format is available in the ``mlflow/java`` package. For more information, see :py:mod:`mlflow.mleap`.
+The ``mleap`` model flavor supports saving Spark models in MLflow format using the 
+`MLeap <http://mleap-docs.combust.ml/>`_ persistence mechanism. MLeap is an inference-optimized 
+format and execution engine for Spark models that does not depend on 
+`SparkContext <https://spark.apache.org/docs/latest/api/python/pyspark.html#pyspark.SparkContext>`_
+to evaluate inputs.
+
+Spark models can be saved with in MLflow format with MLeap by specifying the ``sample_input`` 
+argument of the :py:func:`mlflow.spark.save_model()` or :py:func:`mlflow.spark.log_model()` method
```
Comment (stbof) [inline docs/source/models.rst]: Spark models can be saved with in MLflow format with MLeap -> You can save Spark models in MLflow format with MLeap flavor

File: docs/source/models.rst [hunk 5]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
+to evaluate inputs.
+
+Spark models can be saved with in MLflow format with MLeap by specifying the ``sample_input`` 
+argument of the :py:func:`mlflow.spark.save_model()` or :py:func:`mlflow.spark.log_model()` method 
+(recommended). The :py:mod:`mlflow.mleap` module also defines 
+:py:func:`save_model() <mlflow.mleap.save_model>` and 
+:py:func:`log_model() <mlflow.mleap.log_model>` methods for saving MLeap models in MLflow format, 
+but these methods do not include the ``python_function`` flavor in the models they produce.
+
+A companion module for loading MLflow models with the MLeap flavor format is available
```
Comment (stbof) [inline docs/source/models.rst]: A companion module for loading MLflow models with the MLeap flavor format  -> A companion module for loading MLflow model format  with the MLeap flavor

File: docs/source/models.rst [hunk 6]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
-The ``pytorch`` model flavor enables logging and loading PyTorch models. Model is completely stored in `.pth` format using `torch.save(model)` method. Given a directory containing a saved model, you can log the model to MLflow via ``log_saved_model``. The saved model can then be loaded for inference via ``mlflow.pyfunc.load_pyfunc()``. For more information, see :py:mod:`mlflow.pytorch`.
+The ``pytorch`` model flavor enables logging and loading PyTorch models.
+
+The :py:mod:`mlflow.pytorch` module defines utilities for saving and loading MLflow models with the
+``pytorch`` flavor. The :py:func:`mlflow.pytorch.save_model()` and
+:py:func:`mlflow.pytorch.log_model()` methods can be used to save PyTorch models in MLflow format;
+both of these functions use the
+`torch.save() <https://pytorch.org/docs/stable/torch.html#torch.save>`_ method to serialize PyTorch
+models. Additionally, the :py:func:`mlflow.pytorch.load_model()` method can be used to load MLflow
+models with the ``pytorch`` flavor as PyTorch model objects. Finally, models produced by
```
Comment (stbof) [inline docs/source/models.rst]: the :py:func:`mlflow.pytorch.load_model()` method can be used to load MLflow
models with the ``pytorch`` flavor -> You can use the :py:func:`mlflow.pytorch.load_model()` method to load MLflow models with the ``pytorch`` flavor

File: docs/source/models.rst [hunk 7]
```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
...
-external dependencies. It saves and loads models using Python's pickle module and also generates a valid
-``python_function`` flavor model. For more information, see :py:mod:`mlflow.sklearn`.
+The ``sklearn`` model flavor provides an easy-to-use interface for saving and loading scikit-learn
+models. The :py:mod:`mlflow.sklearn` module defines 
+:py:func:`save_model() <mlflow.sklearn.save_model>` and 
+:py:func:`log_model() <mlflow.sklearn.log_model>` functions that save scikit-learn models in
+MLflow format, using either Python's pickle module (Pickle) or CloudPickle for model serialization.
+These functions produce MLflow models with the ``python_function`` flavor, allowing them to
+be loaded as generic Python functions for inference via :py:func:`mlflow.pyfunc.load_pyfunc()`.
+Finally, the :py:func:`mlflow.sklearn.load_model()` method can be used to load MLflow models with
```
Comment (stbof) [inline docs/source/models.rst]: allowing them to be loaded as generic Python functions -> allowing you to load them as generic Python functions

**Comment by stbof** [comment]
In general try to use active voice "You can load", rather than "can be loaded".

File: docs/source/models.rst (hunk 1 above)
Comment (dbczumar) [reply docs/source/models.rst]: Fixed!

File: docs/source/models.rst (hunk 2 above)
Comment (dbczumar) [reply docs/source/models.rst]: Fixed!

File: docs/source/models.rst (hunk 3 above)
Comment (dbczumar) [reply docs/source/models.rst]: Done!

File: docs/source/models.rst (hunk 4 above)
Comment (dbczumar) [reply docs/source/models.rst]: Changed to active voice!

File: docs/source/models.rst (hunk 5 above)
Comment (dbczumar) [reply docs/source/models.rst]: Removed `format`. Thanks!

File: docs/source/models.rst (hunk 6 above)
Comment (dbczumar) [reply docs/source/models.rst]: Fixed!

File: docs/source/models.rst (hunk 7 above)
Comment (dbczumar) [reply docs/source/models.rst]: Changed to active voice!

**Comment by dbczumar** [comment]
@stbof Thank you for the review and for providing examples for how to correct each issue. I've updated the PR to use active voice throughout. Let me know how it looks!

**Comment by stbof** [comment]
LGTM.

**Pull request merged**